src/compiler/scala/tools/nsc/transform/Erasure.scala

```diff
@@ -411,6 +414,15 @@ abstract class Erasure extends InfoTransform
     }
     val throwsArgs = sym0.annotations flatMap ThrownException.unapply
     if (needsJavaSig(sym0, info, throwsArgs)) {
+      val mixinTParams: Set[Symbol] = if (sym0.hasFlag(MIXEDIN)) info.typeParams.toSet else Set.empty
+      if (mixinTParams.nonEmpty) {
```

Comment:
As a matter of efficiency, maybe we can avoid creating the `set` at all. The condition would start as follows: 
```suggestion
      if (sym0.hasFlag(MIXEDIN) && ! info.typeParams.isEmpty) {
```
Then, instead of `!mixinTparams(sym)`, we could just write `info.typeParams.contains(sym)`.

Reply:
Makes sense! I added a commit that implements this and factors out initialization of the set to a separate method.